robot-db/src/main/java/domain/Book.java

```diff
@@ -93,4 +96,35 @@ public String toString () {
                 .append(description)
                 .append("}").toString();
     }
-}
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17,31)
+                .append(title)
+                .append(newPrice)
+                .append(oldPrice)
+                .append(currency)
+                .append(year)
+                .append(link)
+                .append(description)
+                .append(printHouse)
+                .append(authors)
+                .append(category)
+                .toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object bookObj) {
+        if (this == bookObj)
+            return true;
+        if (bookObj == null || getClass() != bookObj.getClass())
+            return false;
+        Book book = (Book)bookObj;
+        return title.equals(book.title)
+                && abs(newPrice - book.newPrice) < 1e-10 && abs(oldPrice - book.oldPrice) < 1e-10
+                && currency.equals(book.currency) && year == book.year
+                && link.equals(book.link) && description.equals(book.description)
+                && printHouse.equals(book.printHouse) && authors.equals(book.authors)
+                && category == book.category;
+    }
+}
```

Comment:
I know that I can split it into multiple lines using a helper boolean variable but I am not sure if it has a sense. 
When the false is found int the expression, next boolean values are not calculated (power of && operator :smile_cat: ). I think it is much faster than using helper variable, but I can test this later.
The drop of performance would be caused by & operator in the case when there are many false values in the expression.
What can I do is to move abs(...) to the end of the expression. JVM will not have to calculate difference, absolute value and compare when there was a false value earlier.